### PR TITLE
docs(*): improve timeout page

### PR DESCRIPTION
**Pull request by lobkovilya**
Don't merge to `master` until 1.7.0 release
Signed-off-by: Ilya Lobkov <ilya.lobkov@konghq.com>

### CODE REVIEW

**Comment by lahabana** [comment]
I have one piece of doc to push and then we'll be able to have 1.7 folder

**Comment by lahabana** [comment]
Are we backporting this in the end?

File: docs/docs/dev/policies/timeout.md [hunk 1]
```diff
@@ -8,7 +8,52 @@ Specify the proxy to configure with the `sources` selector, and the outbound con
 
 The policy lets you configure timeouts for `HTTP`, `GRPC`, and `TCP` protocols.
```
Comment (lahabana) [inline docs/docs/dev/policies/timeout.md:9]: Worth mentioning protocol conf needs to be set in the config and link to the appropriate page.

File: docs/docs/dev/policies/timeout.md [hunk 2]
```diff
@@ -8,7 +8,52 @@ Specify the proxy to configure with the `sources` selector, and the outbound con
...
 
-## Example
+## Configuration
+
+Timeouts applied when communicating with services of **any protocol**:
+
+Field: **connectTimeout**<br>
+Description: time to establish a connection<br>
+Default value: 10s<br>
+Envoy conf: Cluster
```
Comment (lahabana) [inline docs/docs/dev/policies/timeout.md:18]: Maybe worth linking these to the Envoy doc?

File: docs/docs/dev/policies/timeout.md [hunk 3]
```diff
@@ -8,7 +8,52 @@ Specify the proxy to configure with the `sources` selector, and the outbound con
...
+
+Timeouts applied when communicating with **TCP** services:
+
+Field: **tcp.idleTimeout**<br>
+Description: period in which there are no bytes sent or received 
+on either the upstream or downstream connection<br>
+Default value: disabled<br>
+Envoy conf: TCPProxy
+
+Timeouts applied when communicating with **HTTP**, **HTTP2** or **GRPC** services:
```
Comment (lahabana) [inline docs/docs/dev/policies/timeout.md:28]: As http is tcp what's the value of tcp.idleTimeout in this case just not used?

File: docs/docs/dev/policies/timeout.md [hunk 4]
```diff
@@ -17,30 +62,23 @@ apiVersion: kuma.io/v1alpha1
...
-      idleTimeout: 1h
-    grpc:
-      # 'stream_idle_timeout' on HttpConnectionManager, disabled by default
-      streamIdleTimeout: 5m
-      # 'max_stream_duration' on Cluster, disabled by default
-      maxStreamDuration: 30m
+      streamIdleTimeout: 30m
+      maxStreamDuration: 0s
 ```
 We will apply the configuration with `kubectl apply -f [..]`.
```
Comment (lahabana) [inline docs/docs/dev/policies/timeout.md:83]: Want to remove this it's just the default policy

File: docs/docs/dev/policies/timeout.md [hunk 5]
```diff
@@ -49,32 +87,32 @@ We will apply the configuration with `kubectl apply -f [..]`.
...
     idleTimeout: 1h
-  grpc:
-    # 'stream_idle_timeout' on HttpConnectionManager, disabled by default
-    streamIdleTimeout: 5m
-    # 'max_stream_duration' on Cluster, disabled by default
-    maxStreamDuration: 30m
+    streamIdleTimeout: 30m
+    maxStreamDuration: 0s
 ```
 We will apply the configuration with `kumactl apply -f [..]` or via the [HTTP API](../../reference/http-api).
```
Comment (lahabana) [inline docs/docs/dev/policies/timeout.md:107]: Same remove this line

File: docs/docs/dev/policies/timeout.md [hunk 6]
```diff
@@ -88,3 +126,89 @@ Request timeouts are configured on the Envoy routes and may select a different T
...
+  idleTimeout: 2h
+http:
+  requestTimeout: 0s
+  idleTimeout: 2h
+  streamIdleTimeout: 1h
+  maxStreamDuration: 0s
+```
+
+These timeouts are either disabled or 2 times bigger than outbound default values
+to not interfere with the user's smaller values.
```
Comment (lahabana) [inline docs/docs/dev/policies/timeout.md:147]: I'm not sure that's useful information you might just want to say it differently when you offer users to modify it

File: docs/docs/dev/policies/timeout.md [hunk 7]
```diff
@@ -88,3 +126,89 @@ Request timeouts are configured on the Envoy routes and may select a different T
 
 Mesh configures an idle timeout on the HTTPConnectionManager, but doesn’t consistently use the Timeout policy values for this, so the semantica are ambiguous.
 There’s no policy that configures the idle timeout for downstream connections to the Gateway.
+
+## Inbound timeouts
+
+Current policy configures timeouts only on the outbound side.
```
Comment (lahabana) [inline docs/docs/dev/policies/timeout.md:132]:
```suggestion
Currently, there is no policy to set inbound timeouts.
```

File: docs/docs/dev/policies/timeout.md [hunk 8]
```diff
@@ -88,3 +126,89 @@ Request timeouts are configured on the Envoy routes and may select a different T
...
+  idleTimeout: 2h
+  streamIdleTimeout: 1h
+  maxStreamDuration: 0s
+```
+
+These timeouts are either disabled or 2 times bigger than outbound default values
+to not interfere with the user's smaller values.
+
+Despite there is no policy at that moment to modify inbound timeouts,
+all values could be changed using ProxyTemplate:
```
Comment (lahabana) [inline docs/docs/dev/policies/timeout.md:150]: "If you still need to change inbound timeouts you can use a ProxyTemplate" (please link to the PT policy doc)

File: docs/docs/dev/policies/timeout.md [hunk 9]
```diff
@@ -88,3 +126,89 @@ Request timeouts are configured on the Envoy routes and may select a different T
...
+        value: |
+          name: envoy.filters.network.http_connection_manager
+          typedConfig:
+            '@type': type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
+            streamIdleTimeout: 0s # disable http.streamIdleTimeout 
+            common_http_protocol_options: 
+              idle_timeout: 0s # disable http.idleTimeout
+```
+:::
+::::
```
Comment (lahabana) [inline docs/docs/dev/policies/timeout.md:209]: May here add a tip:

" You should always be careful and set the inbound timeout to be larger than the client's outbound timeouts. An easy rule of thumb is to set the inbound timeouts to be double the size of the client's outbound timeouts."

**Review by lahabana — COMMENTED** [review]
Thanks for the clarification on this page! Timeouts are tricky and it's great to see in-depth info on them. @sentinelleader would love to have your eyes on this :)

File: docs/docs/dev/policies/timeout.md (hunk 6 above)
Comment (lobkovilya) [reply docs/docs/dev/policies/timeout.md]: I think it makes sense to simply delete these lines, conf just above is self-explanatory

**Review by lahabana — APPROVED** [review]

**Pull request merged**